### PR TITLE
Minor performance tweak for purge archive for deleted sites

### PR DESCRIPTION
**Pull request by tsteur**
Not expecting a big change in performance here. But generally it's better to have two simple queries that can be fully read from index instead of joining the tables etc. The join should be straight forward though as well because it should be also created based on the index, however the two individual queries should be still faster.

Also another tweak is that for deleted sites we  don't need to look at `ts_archived`. This actually does make things quite a bit faster.

### CODE REVIEW

File: tests/PHPUnit/Integration/Archive/ArchivePurgerTest.php [hunk 1]
```diff
@@ -116,8 +116,8 @@ public function test_purgeNoSiteArchives_PurgesAllNoSiteArchives()
 
         //There are 5 rows for website #3. We leave the other two because they're before our purge threshold.
         $deletedRowCount = $this->archivePurger->purgeDeletedSiteArchives($this->january);
-        $this->assertEquals(3 * RawArchiveDataWithTempAndInvalidated::ROWS_PER_ARCHIVE, $deletedRowCount);
-        self::$fixture->assertArchivesDoNotExist(array(3, 7, 10), $this->january);
+        $this->assertEquals(5 * RawArchiveDataWithTempAndInvalidated::ROWS_PER_ARCHIVE, $deletedRowCount);
```
Comment (tsteur) [inline tests/PHPUnit/Integration/Archive/ArchivePurgerTest.php]: We are now expecting more results as for deleted sites we don't need to look at ts_archived.

File: core/DataAccess/Model.php [hunk 2]
```diff
@@ -319,14 +319,30 @@ public function getSitesWithInvalidatedArchive($numericTable)
...
+        $deletedSites = array_diff($idSitesUsed, $idSitesExisting);
+
+        if (empty($deletedSites)) {
+            return array();
+        }
+        $deletedSites = array_values($deletedSites);
+
+        $sql = "SELECT DISTINCT idarchive FROM " . $archiveTableName . " WHERE idsite IN (".Common::getSqlStringFieldsArray($deletedSites).")";
+
+        $rows = Db::getReader()->fetchAll($sql, $deletedSites);
```
Comment (diosmosis) [inline core/DataAccess/Model.php]: If there are hundreds or thousands of deleted sites, will this be a problem? (since we're using placeholders)

Comment (tsteur) [reply core/DataAccess/Model.php]: I reckon it's unlikely this happens but will change it indeed 👍 doesn't hurt

Comment (tsteur) [reply core/DataAccess/Model.php]: Will also apply this to the `deleteArchiveIds` method that is in the same model

Comment (tsteur) [reply core/DataAccess/Model.php]: Done 👍

**Pull request merged**